src/SonarScanner.MSBuild.Common/MsBuildPathSettings.cs

```diff
@@ -32,7 +32,18 @@ public class MsBuildPathSettings : IMsBuildPathsSettings
         /// errors from the Integration targets in case we detect we are running under
         /// not supported MSBuild.
         /// </summary>
-        private readonly string[] msBuildVersions = new[] { "4.0", "10.0", "11.0", "12.0", "14.0", "15.0", };
+        /// <remarks>
+        /// From MSBuild 16.0 onwards, there will no longer be a version-specific folder. Instead,
+        /// all versions of MSBuild use "Current".
+        /// This means that if we ever need to provide version-specific behaviour in the ImportBefore
+        /// targets, we will need to put all of the behaviours in a single file, and use the
+        /// property $(MSBuildAssemblyVersion) to determine version of MSBuild is executing.
+        /// See the following tickest for more info:
+        /// * https://github.com/SonarSource/sonar-scanner-msbuild/issues/676
+        /// * https://github.com/Microsoft/msbuild/issues/3778
+        /// * https://github.com/Microsoft/msbuild/issues/4149 (closed as "Won't fix")
+        /// </remarks>
+        private readonly string[] msBuildVersions = new[] { "4.0", "10.0", "11.0", "12.0", "14.0", "15.0", "Current" };
```

Comment:
The only change required in the product code is to add "Current" to the list of MSBuild folders.